pkg/virt-handler/vm.go

```diff
@@ -295,12 +295,20 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 			for _, domainInterface := range domain.Spec.Devices.Interfaces {
 				interfaceMAC := domainInterface.MAC.MAC
 				var newInterface v1.VirtualMachineInstanceNetworkInterface
+				var isMasqueredeInterface bool = false
 
+				for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
+					if iface.Name == domainInterface.Alias.Name && iface.Masquerade != nil {
+						isMasqueredeInterface = true
+					}
+				}
 				if existingInterface, exists := existingInterfaceStatusByName[domainInterface.Alias.Name]; exists {
 					// Reuse previously calculated interface from vmi.Status.Interfaces, updating the MAC from domain.Spec
 					// Only interfaces defined in domain.Spec are handled here
 					newInterface = existingInterface
-					newInterface.MAC = interfaceMAC
+					if !isMasqueredeInterface {
```

Comment:
Add an explanation similar to the MAC one.

Reply:
Added